bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/flamingos2/F2ScienceAreaGeometry.scala

```diff
@@ -1,57 +1,52 @@
+import edu.gemini.spModel.obs.context.ObsContext
+
+import java.awt.Shape
+import java.awt.geom.{Rectangle2D, Area, Ellipse2D}
 
-import scala.collection.JavaConverters._
-import scalaz.Scalaz._
 import scalaz._
+import Scalaz._
 
-class F2ScienceAreaGeometry(inst0: Flamingos2) extends ScienceAreaGeometry {
-  import F2ScienceAreaGeometry._
+object F2ScienceAreaGeometry extends ScienceAreaGeometry {
 
-  override def geometry: List[Shape] = {
-    Option(inst0).fold(List[Shape]()) { inst =>
-      lazy val plateScale = inst.getLyotWheel.getPlateScale
-      lazy val scienceAreaWidth = {
-        val scienceAreaArray = inst.getScienceArea
-        scienceAreaArray(0)
-      }
-      inst.getFpu match {
-        case Flamingos2.FPUnit.FPU_NONE    => List(imagingFOV(plateScale))
-        case Flamingos2.FPUnit.CUSTOM_MASK => List(mosFOV(plateScale))
-        case _ if inst.getFpu.isLongslit   => List(longSlitFOV(plateScale, scienceAreaWidth))
-        case _                             => Nil
-      }
-    }
-  }
+  override def unadjustedGeometry(ctx: ObsContext): Option[Shape] =
+    ctx.getInstrument match {
+      case f2: Flamingos2 =>
+        val plateScale       = f2.getLyotWheel.getPlateScale
+        val scienceAreaWidth = scienceAreaDimensions(f2)._1
+        f2.getFpu match {
+          case Flamingos2.FPUnit.FPU_NONE    => Some(imagingFOV(plateScale))
+          case Flamingos2.FPUnit.CUSTOM_MASK => Some(mosFOV(plateScale))
+          case _ if f2.getFpu.isLongslit     => Some(longSlitFOV(plateScale, scienceAreaWidth))
+          case _                             => None
+        }
 
-  // We need to override this due to type system.
-  override def geometryAsJava: edu.gemini.shared.util.immutable.ImList[Shape] =
-    DefaultImList.create(geometry.asJava)
+      case _              => None
+    }
 
-  def scienceAreaDimensions: (Double, Double) = {
-    import LyotWheel._
-    Option(inst0).map { inst => inst.getLyotWheel match {
+  def scienceAreaDimensions(f2: Flamingos2): (Double, Double) =
+    f2.getLyotWheel match {
       case OPEN | HIGH | LOW =>
-        import FPUnit._
-        lazy val plateScale = inst.getLyotWheel.getPlateScale
-        inst.getFpu match {
+        val plateScale = f2.getLyotWheel.getPlateScale
+        f2.getFpu match {
           case FPU_NONE =>
             val size = ImagingFOVSize * plateScale
             (size, size)
           case CUSTOM_MASK =>
             (MOSFOVWidth * plateScale, ImagingFOVSize * plateScale)
           case fpu if fpu.isLongslit =>
-            (fpu.getSlitWidth * inst.getLyotWheel.getPixelScale, LongSlitFOVHeight * plateScale)
+            (fpu.getSlitWidth * f2.getLyotWheel.getPixelScale, LongSlitFOVHeight * plateScale)
         }
       case _ => (0.0, 0.0)
-    }}.getOrElse((0.0, 0.0))
-  }
+    }
+
+  def javaScienceAreaDimensions(f2: Flamingos2): Array[Double] =
+    scienceAreaDimensions(f2) |> { case (w, h) => Array(w, h) }
```

Comment:
Why did you use an `unsafeTap` here could you have done `Array(scienceAreaDimensions(f2)._1, scienceAreaDimensions(f2)._2)